### PR TITLE
add Junit tests

### PR DESCRIPTION
**Pull request by AbdelSBIHI**

### CODE REVIEW

File: .tern-project [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (darkjeffbr) [inline .tern-project]: Please, do not commit configuration files.

File: pom.xml [hunk 2]
```diff
@@ -50,6 +50,14 @@
...
 			<version>2.0.8</version>
 		</dependency>
+
+	<!-- https://mvnrepository.com/artifact/org.mockito/mockito-core -->
+	<dependency>
+		<groupId>org.mockito</groupId>
+		<artifactId>mockito-core</artifactId>
+		<version>3.7.7</version>
+		<scope>test</scope>
+	</dependency>
```
Comment (darkjeffbr) [inline pom.xml]: I miss here the junit dependency.

File: src/main/java/com/everis/control/CarService.java [hunk 3]
```diff
@@ -1,81 +1,61 @@
...
-    	LOGGER.info("Car's Id chosen for delete: " + id);
-    	em.remove(car);
-    	LOGGER.info("Deleted Car: " + car);	
-    }
+	private CarDao carDao = new CarDao();
+
+	/**
+	 * Method to get a list of Car Entity available
+	 */
+
```
Comment (darkjeffbr) [inline src/main/java/com/everis/control/CarService.java:16]: Nice that you've wrapped the code in new dao class, but why did you remove the logs ?

File: src/test/java/AllTests.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+import org.junit.runners.Suite.SuiteClasses;
+
+import com.everis.boundary.CarResourcesTest;
+import com.everis.control.CarServiceTest;
+
+@RunWith(Suite.class)
+@SuiteClasses({CarServiceTest.class, CarResourcesTest.class })
```
Comment (darkjeffbr) [inline src/test/java/AllTests.java:9]: This class is not necessary, maven and the IDE automatically run all the tests for us 😄

File: src/main/java/com/everis/entity/Car.java [hunk 5]
```diff
@@ -56,6 +56,15 @@
 	public Car() {
 		super();
 	}   
+	
+	public Car(@NotNull(message = "Brand can't be null") String brand, Date registration,
```
Comment (darkjeffbr) [inline src/main/java/com/everis/entity/Car.java:60]: Why you've added the bean validations annotations here ?

File: src/test/java/com/everis/boundary/CarResourcesTest.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+		try {
+			doNothing().when(this.carService).createCar(car);
+			Response responseTest = this.carResource.createCar(car);
+			assertEquals(Status.CREATED.getStatusCode(), responseTest.getStatus());
+			assertEquals(car, responseTest.getEntity());
+			
+		
+		} catch (Exception e) {
+			fail("create car test failed");
+		}
```
Comment (darkjeffbr) [inline src/test/java/com/everis/boundary/CarResourcesTest.java]: It is not needed to catch the exception and call the junit fail method. If an exception happens the test will fail automatically.

File: src/test/java/com/everis/boundary/CarResourcesTest.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import com.everis.control.CarService;
+import com.everis.entity.Car;
+
+public class CarResourcesTest {
```
Comment (darkjeffbr) [inline src/test/java/com/everis/boundary/CarResourcesTest.java:25]: Please, add also the negative tests.

File: src/test/java/com/everis/control/CarServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+			Mockito.when(carDao.findAll()).thenReturn(allcars);
+			final List<Car> cars = carService.getCars();
+			Assert.assertEquals(allcars, cars);
+		} catch (Exception e) {
+			Assert.fail();
+		}
+	}
+
+	@Test
+	public void testGetCar() {
```
Comment (darkjeffbr) [inline src/test/java/com/everis/control/CarServiceTest.java:39]: Is this test working ? Did you try to run it ?

File: src/test/java/com/everis/control/CarServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+import com.everis.entity.Car;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CarServiceTest {
+
+	@InjectMocks
+	CarService carService;
+
+	@Mock
+	private CarDao carDao;
```
Comment (darkjeffbr) [inline src/test/java/com/everis/control/CarServiceTest.java:26]: Please, also add unit tests for the CarDao class.

**Review by darkjeffbr — CHANGES_REQUESTED** [review]

File: src/test/java/com/everis/control/CarServiceTest.java (hunk 8 above)
Comment (AbdelSBIHI) [reply src/test/java/com/everis/control/CarServiceTest.java:39]: yes its works

File: src/main/java/com/everis/entity/Car.java (hunk 5 above)
Comment (AbdelSBIHI) [reply src/main/java/com/everis/entity/Car.java:60]: i just needed i constructor with parameters so i generated by eclipse shortcuts

File: src/main/java/com/everis/control/CarService.java (hunk 3 above)
Comment (AbdelSBIHI) [reply src/main/java/com/everis/control/CarService.java:16]: i think i have now the interceptor for logging

**Review by darkjeffbr — APPROVED** [review]
